File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/search/model/tv/SearchTvShow.java
package dev.reuss.tmdb.domain.search.model.tv;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

import java.util.List;

/**
 * TV show returned by TMDB TV search.
 *
 * @param adult            whether the TV show is marked as adult
 * @param backdropPath     backdrop image path
 * @param genreIds         genre ids
 * @param id               TMDB TV show id
 * @param originCountry    origin countries
 * @param originalLanguage original language code
 * @param originalName     original TV show name
 * @param overview         TV show overview
 * @param popularity       TV show popularity
 * @param posterPath       poster image path
 * @param firstAirDate     first air date
 * @param name             localized TV show name
 * @param voteAverage      vote average
 * @param voteCount        vote count
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record SearchTvShow(
        boolean adult,

        @JsonProperty("backdrop_path")
        String backdropPath,

        @JsonProperty("genre_ids")
        List<Integer> genreIds,

        int id,

        @JsonProperty("origin_country")
        List<String> originCountry,

        @JsonProperty("original_language")
        String originalLanguage,

        @JsonProperty("original_name")
        String originalName,

        String overview,

        double popularity,

        @JsonProperty("poster_path")
        String posterPath,

        @JsonProperty("first_air_date")
        String firstAirDate,

        String name,

        @JsonProperty("vote_average")
        double voteAverage,

        @JsonProperty("vote_count")
        int voteCount
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/search/model/tv/SearchTvShow.kt
package dev.reuss.tmdb.domain.search.model.tv

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * TV show returned by TMDB TV search.
 *
 * @property adult whether the TV show is marked as adult
 * @property backdropPath backdrop image path
 * @property genreIds genre ids
 * @property id TMDB TV show id
 * @property originCountry origin countries
 * @property originalLanguage original language code
 * @property originalName original TV show name
 * @property overview TV show overview
 * @property popularity TV show popularity
 * @property posterPath poster image path
 * @property firstAirDate first air date
 * @property name localized TV show name
 * @property voteAverage vote average
 * @property voteCount vote count
 */
@JvmRecord
data class SearchTvShow(
    @all:JsonProperty("adult")
    val adult: Boolean,

    @all:JsonProperty("backdrop_path")
    val backdropPath: String?,

    @all:JsonProperty("genre_ids")
    val genreIds: List<Int> = emptyList(),

    @all:JsonProperty("id")
    val id: Int,

    @all:JsonProperty("origin_country")
    val originCountry: List<String> = emptyList(),

    @all:JsonProperty("original_language")
    val originalLanguage: String?,

    @all:JsonProperty("original_name")
    val originalName: String?,

    @all:JsonProperty("overview")
    val overview: String?,

    @all:JsonProperty("popularity")
    val popularity: Double,

    @all:JsonProperty("poster_path")
    val posterPath: String?,

    @all:JsonProperty("first_air_date")
    val firstAirDate: String?,

    @all:JsonProperty("name")
    val name: String?,

    @all:JsonProperty("vote_average")
    val voteAverage: Double,

    @all:JsonProperty("vote_count")
    val voteCount: Int
) : TmdbModel